tempodb/blocklist/poller.go

```diff
@@ -109,23 +117,41 @@ type Poller struct {
 
 	sharder JobSharder
 	logger  log.Logger
+
+	tenantQueues     *flushqueues.ExclusiveQueues
+	tenantQueuesDone sync.WaitGroup
+	tenantsPolled    map[string]time.Time
+
+	perTenantChan          chan *PerTenant
+	perTenantCompactedChan chan *PerTenantCompacted
+	perTenantErrChan       chan error
+
+	blocklist *List
 }
 
 // NewPoller creates the Poller
-func NewPoller(cfg *PollerConfig, sharder JobSharder, reader backend.Reader, compactor backend.Compactor, writer backend.Writer, logger log.Logger) *Poller {
+func NewPoller(cfg *PollerConfig, sharder JobSharder, reader backend.Reader, compactor backend.Compactor, writer backend.Writer, logger log.Logger, blocklist *List) *Poller {
 	return &Poller{
 		reader:    reader,
 		compactor: compactor,
 		writer:    writer,
 
-		cfg:     cfg,
-		sharder: sharder,
-		logger:  logger,
+		cfg:           cfg,
+		sharder:       sharder,
+		logger:        logger,
+		tenantQueues:  flushqueues.New(int(cfg.TenantPollConcurrency), metricTenantQueueLength),
+		tenantsPolled: make(map[string]time.Time, 1000),
+
+		perTenantChan:          make(chan *PerTenant, 1000),
```

Comment:
The static `1000` here is the part of this PR I'm least comfortable with.  I'm not sure how to size this.

Reply:
Turns out if this is 0 the code still seems to work.  Continuing to test.